doc/directory/file.rst

```diff
@@ -35,3 +35,19 @@ with explicit grouping:
 
     In [2]: mapping = {key : value for key, value in zip( keys,values) }
        ...: mapping
+
+with testcode:
+
+.. testsetup::
+
+   x = 'X'
+   y =  "Y"
+
+.. testcode::
+
+   assert x ==  x
+   assert x !=  y
```

Comment:
while this will be reformatted, the two spaces are not really easy to see. Maybe something like
```suggestion
   assert x==x
   assert x !=  y
```
would be more obvious?